code/datums/spells/devil.dm

```diff
@@ -0,0 +1,211 @@
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/Destroy()
+	if(pitchfork)
+		qdel(pitchfork)
+		pitchfork = null
+	return ..()
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/greater
+	pitchfork_type = /obj/item/weapon/twohanded/pitchfork/demonic/greater
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/ascended
+	pitchfork_type = /obj/item/weapon/twohanded/pitchfork/demonic/ascended
+
+/obj/effect/proc_holder/spell/targeted/summon_pitchfork/violin
+	pitchfork_type = /obj/item/device/violin/golden //Oh my god this is the most ghetto code reuse ever.
+	invocation_type = "whisper"
+	invocation = "I aint have this much fun since Georgia."
+	action_icon_state = "golden_violin"
+	name = "Summon golden violin"
+
+
+/obj/effect/proc_holder/spell/targeted/summon_contract
+	name = "Summon infernal contract"
+	desc = "Skip making a contract by hand, just do it by magic."
+	invocation_type = "whisper"
+	invocation = "Just sign on the dotted line."
+	include_user = 0
+	range = 5
+	clothes_req = 0
+
+	school = "conjuration"
+	charge_max = 150
+	cooldown_min = 10
+	action_icon_state = "spell_default"
+	action_background_icon_state = "bg_demon"
+
+/obj/effect/proc_holder/spell/targeted/summon_contract/cast(list/targets, mob/user = usr)
+	for(var/mob/living/carbon/C in targets)
+		if(C.mind && user.mind)
+			if(C.stat == DEAD)
+				if(user.drop_item())
+					var/obj/item/weapon/paper/contract/infernal/revive/contract = new(user.loc, C.mind, user.mind)
+					user.put_in_hands(contract)
+			else
+				var/obj/item/weapon/paper/contract/infernal/contract  // = new(user.loc, C.mind, contractType, user.mind)
+				var/contractTypeName = input(user, "What type of contract?") in list ("Power", "Wealth", "Prestige", "Magic", "Knowledge")
+				switch(contractTypeName)
+					if("Power")
+						contract = new /obj/item/weapon/paper/contract/infernal/power(C.loc, C.mind, user.mind)
+					if("Wealth")
+						contract = new /obj/item/weapon/paper/contract/infernal/wealth(C.loc, C.mind, user.mind)
+					if("Prestige")
+						contract = new /obj/item/weapon/paper/contract/infernal/prestige(C.loc, C.mind, user.mind)
+					if("Magic")
+						contract = new /obj/item/weapon/paper/contract/infernal/magic(C.loc, C.mind, user.mind)
+					if("Knowledge")
+						contract = new /obj/item/weapon/paper/contract/infernal/knowledge(C.loc, C.mind, user.mind)
+				C.put_in_hands(contract)
+		else
+			to_chat(user,"<span class='notice'>[C] seems to not be sentient.  You cannot summon a contract for them.</span>")
+
+
+/obj/effect/proc_holder/spell/dumbfire/fireball/hellish
+	name = "Hellfire"
+	desc = "This spell launches hellfire at the target."
+
+	school = "evocation"
+	charge_max = 80
+	clothes_req = 0
+	invocation = "Your very soul will catch fire!"
+	invocation_type = "shout"
+	range = 2
+	action_background_icon_state = "bg_demon"
+
+/obj/effect/proc_holder/spell/turf/fireball/infernal/cast(var/turf/T)
+	explosion(T, -1, -1, -1, 4, 0, flame_range = 5)
```

Comment:
I'm not sure if this proc will ever be called, since you don't set `proj_type` on `fireball/hellish`